src/fileioc/fileioc.asm

```diff
@@ -1573,6 +1600,67 @@ util_set_offset:
 	ld	(hl), bc
 	ret
 
+util_Arc_Unarc: ;properly handle garbage collects :P
+	call _ChkFindSym
+	push hl
+	call _ChkInRAM
+	pop hl
+	jr nz,.arc_unarc ;if the file is already in archive, we won't trigger a gc
+	call _LoadDEInd_s
+	ld hl,12
+	add hl,de
+	call _FindFreeArcSpot ;check if we will trigger a gc
+	jr nz,.arc_unarc ;gc will not be triggered
+	ld a,(mpLcdCtrl)
+	cp a,lcdBpp16
+	jr z,.arc_unarc ;already in OS 16bpp graphics mode
+	push af ;save lcd mode
+	call _ClrLCDFull
+	call _DrawStatusBar
+	ld a,lcdBpp16
+	ld (mpLcdCtrl),a
```

Comment:
Should become a call to the default, no-op pre-GC handler.

Reply:
Default handler now points to a ret instruction, added pre-GC handler.